RELEASE.md

```diff
@@ -0,0 +1,160 @@
+- Configure a GitHub token for release scripts to use.
+  - https://github.com/settings/tokens
+  - Select the following permissions: 
+    - `admin:org`
+    - `public_repo`
+    - `read:user`
+  - Ensure your token is available to scripts, e.g. `export GH_API_TOKEN={YOUR-TOKEN}` or similar.
+- Get WPORG plugin svn commit access - ask a team member.
+
+_Outcome_: __You are equipped to ship a release!__
+
+## Release process
+### Lead-up to release
+#### Ensure release development is complete
+- For _major_ and _minor_ releases, we use ZenHub.
+- For _patch_ releases, we use GitHub milestones - [example](https://github.com/woocommerce/woocommerce-gutenberg-products-block/milestone/41).
+- Ensure all issues/PRs intended for this release are merged, closed and linked to release.
+- All PRs should have changelog entry, or `skip-changelog` tag.
+- Check with the team to confirm any outstanding or in progress work.
+
+Note: changelog should be formatted like this in PR description. Note the preceding `> ` - this is required by changelog script.
+
+```md
+### Changelog
+
+> bug: Fix bug in Safari and other Webkit browsers that was causing the All Products block to show 0 results when resetting the sort value.
+```
+
+_Outcome_: __Team is aware of release and in agreement about what fixes & features are included.__
+
+#### Ensure release branch includes all relevant fixes
+- Make release branch if needed.
+  - For _major_ and _minor_ releases, create branch: `release/X.X`.
+  - For _patch_ releases, the branch should already exist.
+- Update your local checkout to the tip of the release branch.
+- For _patch_ releases, cherry pick relevant PRs into the release branch:
+  - If PR is already labelled `status: cherry-picked 🍒` then continue to next PR.
+  - Ideally, use GitHub Hub to cherry pick the PR - `hub am -3 {http://URL-TO-PR}`. 
+  - If there are serious conflicts or extensive differences between `master` and release branch, you may need to take more care:
+    - Manually cherry pick individual commits using git - `git cherry-pick {COMMIT-HASH}`.
+    - Or in some cases, manually craft a new PR with appropriate changes, targeting release branch.
+  - Push the release branch to origin (so changes are in GitHub repo).
+  - Label the PR: `status: cherry-picked 🍒`.
+
+_Outcome_: __Release branch has all relevant changes merged & pushed.__
+
+### Prepare release
+#### Ensure release branch readme is up to date
+- Run changelog script `npm run changelog` to get changelog txt for readme. Changelog content will be output to screen by script.
+- Add changelog section for release, e.g. [`= 2.5.11 - 2020-01-20 =`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/commit/74a41881bfa456a2167a52aaeb4871352255e328).
+- Copy-paste the changelog content into `readme.txt`.
+- Make any other changes to readme as needed - e.g. support versions changing, new blocks. 
+- Push readme changes to release branch on origin repo.
```

Comment:
Suggestion to add something at the end of this line: `If you are confident of your changes, no need to create a PR and wait for approval. You can directly push.`. We had some confusion in the past whether this had to be done via a PR or a direct push.